conans/test/unittests/cli/common_test.py

```diff
@@ -0,0 +1,51 @@
+import textwrap
+from unittest.mock import MagicMock
+
+import pytest
+
+from conan.api.conan_api import ConanAPIV2
+from conan.tools.files import save
+from conans.cli.common import get_profiles_from_args
+from conans.client.cache.cache import ClientCache
+from conans.errors import ConanException
+from conans.test.utils.test_files import temp_folder
+
+
+@pytest.fixture()
+def conan_api():
+    tmp_folder = temp_folder()
+    cache = ClientCache(tmp_folder)
+    save(None, cache.default_profile_path, textwrap.dedent("""\
+        [settings]
+        [options]
+        [tool_requires]
+        [conf]
+    """))
+    return ConanAPIV2(tmp_folder)
+
+
+@pytest.fixture()
+def argparse_args():
+    return MagicMock(
+        profile_build=None,
+        profile_host=None,
+        settings_build=None,
+        settings_host=None,
+        options_build=None,
+        options_host=None,
+        conf_build=None,
+        conf_host=None
+    )
+
+
+@pytest.mark.parametrize("conf_name", [
+    "core.doesnotexist:never",
+    "core:doesnotexist"
+])
+def test_core_confs_not_allowed_via_cli(conan_api, argparse_args, conf_name):
```

Comment:
This test reads complex, too many mocks, etc. If it is a unit test of profile loader, why unittesting the ``get_profiles_from_args`` which is a cli helper, instead of testing the profile loader (that can be directly called by the Python API ProfileAPI without calling that cli helper?

Reply:
Because I wanted to test the function that is used by several commands at the same time. If I test the ProfileLoader I could not ensure that the commands are getting that function so I thought it was better covered so. That was the reason.
Talking about mocks, I don't think a mock for argparse is too much but I can change the test and call only the  ProfileLoader function if you think is good enough.

Reply:
I just added another one for ProfileLoader so, if you want, I can get rid of this one.